plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -437,7 +461,31 @@ private KubernetesListBuilder generateAppResources(List<ImageConfiguration> imag
             String message = ValidationUtil.createValidationMessage(e.getConstraintViolations());
             log.error("ConstraintViolationException: %s", message);
             throw new MojoExecutionException(message, e);
+        } finally {
+            composeUtil.cleanComposeRresources();
+        }
+    }
+
+    private Path checkComposeConfig() {
+        return ifComposeConfigPresent() ? buildComposeFilePath() : lookDefaultComposeConfig();
+    }
+
+    private Path lookDefaultComposeConfig() {
+        File[] composeFiles = null;
+
+        if(composeResourceDir != null && composeResourceDir.exists()) {
+            composeFiles = composeResourceDir.listFiles();
         }
+
+        return composeFiles != null && composeFiles.length > 0 ? composeFiles[0].toPath() : null;
```

Comment:
In case multiple compose files are defined into the `src/main/fabric8-compose` dir, I don't know if it makes sense to try to merge them in some way (this is something that can be discussed). If I'm going to use the first one only, It's better to log a warn, e.g. to tell the user which files are being ignored.